### ISSUE TITLE
[Bug] DeviceCode Verification Url is incorrect for Entra External ID tenants

### ISSUE THREAD

**Issue by Johno-ACSLive** [AzureAD/microsoft-authentication-library-for-dotnet#4666]
### Library version used

4.59.0

### .NET version

Brand new .NET 8 Console application.

### Scenario

PublicClient - desktop app

### Is this a new or an existing app?

This is a new app or experiment

### Issue description and reproduction steps

Create a Entra External ID tenant and follow the remaining instructions. The arcticle [here](https://medium.com/the-new-control-plane/authenticating-with-entra-external-id-for-customers-ciam-with-the-device-code-flow-f006d0ac87ef) provides details on the steps and issues.

The URL provided in devicecode.VerificationUrl is "https://microsoft.com/devicelogin" but when logging in with a local account the follow error is displayed "AADSTS500208: The domain is not a valid login domain for the account type". The article [here](https://medium.com/the-new-control-plane/authenticating-with-entra-external-id-for-customers-ciam-with-the-device-code-flow-f006d0ac87ef) documents the issue.

### Expected behavior

The correct URL for the domain should be provided e.g. "https://tenant_sub_domain.ciamlogin.com/common/oauth2/deviceauth". I'm not sure if this is an issue with Microsoft.Identity.Client or MSAL but I'm using Microsoft.Identity.Client library.

### Identity provider

Microsoft Entra External ID

### Solution and workarounds

N/A - at the moment I have hardcoded the correct Uri in my prototype console application

**Comment by bgavrilMS** [comment]
@Johno-ACSLive - are you saying that the `deviceCode.VerificationUrl` is wrong? And if you replace the value provided by the token issuer the flow works?

**Comment by Johno-ACSLive** [comment]
@bgavrilMS yes if I use the url "https://microsoft.com/devicelogin" as provided by the library I receive the error I mentioned.

If I use the alternate url I mentioned I'm successfully able to authenticate and the user flow works as expected.

**Comment by bgavrilMS** [comment]
CC @SammyO - this looks like a CIAM service issue

**Comment by bgavrilMS** [comment]
@Johno-ACSLive - I had a chat with the CIAM team and they have confirmed that device code flow is not yet ready and we've published samples by mistake. 

Of course, you may continue to use the workaround you found, but we cannot guarantee that it'll work forever. 

Let's keep this work item open until the CIAM service fixes all of this.

CC @SammyO

**Comment by rayluo** [comment]
> ### Issue description and reproduction steps
> 
> Create a Entra External ID tenant and follow the remaining instructions. The arcticle [here](https://medium.com/the-new-control-plane/authenticating-with-entra-external-id-for-customers-ciam-with-the-device-code-flow-f006d0ac87ef) provides details on the steps and issues.
> 
> The URL provided in devicecode.VerificationUrl is "https://microsoft.com/devicelogin" but when logging in with a local account the follow error is displayed "AADSTS500208: The domain is not a valid login domain for the account type". The article [here](https://medium.com/the-new-control-plane/authenticating-with-entra-external-id-for-customers-ciam-with-the-device-code-flow-f006d0ac87ef) documents the issue.

For what it's worth, I used an internal test tenant "msidlabciam2" to test Device Code flow just now. It still fails, but with a different error: `AADSTS399266: Blob grant token received with wrong issuer type.`

**Comment by Johno-ACSLive** [comment]
> @Johno-ACSLive - I had a chat with the CIAM team and they have confirmed that device code flow is not yet ready and we've published samples by mistake.
> 
> Of course, you may continue to use the workaround you found, but we cannot guarantee that it'll work forever.
> 
> Let's keep this work item open until the CIAM service fixes all of this.
> 
> CC @SammyO

@bgavrilMS it seems some of the earlier documentation has been updated to indicate this. All good, I'm happy to use the workaround until the Device Code flow is finalised (presumably nothing should break as it will continue to follow the standards which MSAL will follow).

Good idea to keep this open until it's been resolved. Does the CIAM team have a ballpark estimate on when they think it might be ready?

**Comment by bgavrilMS** [comment]
I don't have a estimate, but it is being worked on. As with all Security / Identity work streams, priorities can change quicky :)

**Comment by nguyenlamlll** [comment]
Hi everyone, I'm sorry for extending this thread here. I'm not sure if I'm experiencing the same/related issue. 

But following the author's workaround, using Authority "tenant.ciamlogin.com/organizations" and "https://tenant.ciamlogin.com/common/oauth2/deviceauth" with `AcquireTokenWithDeviceCode(...)`. It works and I can get basic `scp` in the access token to authorize and use against Graph API endpoints. However, it cannot give me the access/scp to use against the downstream API (e.g., `api://{id}/access_as_user`).

On the other hand, if I use `login.microsoftonline.com` and the url provided by `VerificationUrl`, then it works for both Graph API scopes and downstream API's scopes.
